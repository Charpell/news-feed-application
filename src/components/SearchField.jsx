import React from 'react';
import PropTypes from 'prop-types';

/**
 * Render an input field of text
 * @returns {inputField} search field
 * @param {object} props: params passed from NewsFeed
 */
function SearchField(props) {
  const { search, onSearch } = props;
  return (
    <input
      className="form-control"
      placeholder="Search for your favourite news sources on the go..."
      type="text"
      value={search}
      onChange={onSearch}
    />
  );
}


SearchField.propTypes = {
  search: PropTypes.string,
  onSearch: PropTypes.func
};


export default SearchField;
