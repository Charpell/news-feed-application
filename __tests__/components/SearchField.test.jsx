import React from 'react';
import renderer from 'react-test-renderer';
import SearchField from '../../src/components/SearchField';

describe('SearchField', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<SearchField />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
