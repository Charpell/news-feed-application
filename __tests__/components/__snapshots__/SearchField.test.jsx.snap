// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField should render correctly 1`] = `
<input
  className="form-control"
  onChange={undefined}
  placeholder="Search for your favourite news sources on the go..."
  type="text"
  value={undefined}
/>
`;
